partner_compassion/models/partner_compassion.py

```diff
@@ -533,6 +535,26 @@ def compute_geopoint(self):
             partner.advocate_details_id.write(vals)
         return True
 
+    def update_state_from_zip(self, vals):
+        zip_ = vals.get("zip")
+        country_id = vals.get("country_id") or self.country_id.id
+        domain = [
+            ("name", "=", zip_),
+            ("city_id.country_id", "=", country_id),
+        ]
+        city_zip = self.env["res.city.zip"].search(domain)
+        state_id = set(city_zip.mapped("city_id.state_id.id"))
+        if len(state_id) > 1:
+            raise UserError(_("This ZIP is in multiple states !"))
```

Comment:
Let's not raise an error at this stage and just use the first result (which should never happen BTW):
```suggestion
        city_zip = self.env["res.city.zip"].search(domain, limit=1)
```